gpii/node_modules/flowManager/src/CloudBasedFlowManager.js

```diff
@@ -235,16 +232,30 @@ fluid.defaults("gpii.flowManager.cloudBased.untrustedSettings.handler", {
     }
 });
 
-gpii.flowManager.cloudBased.untrustedSettings.handleRequest = function (onError, request, userToken, deviceString, userService) {
+gpii.flowManager.cloudBased.untrustedSettings.handleRequest = function (onError, request, userToken, deviceString, userService, authGrantFinder) {
     request.privacySettings = {};
 
     // If the userToken has an associated user, then we retrieve the
     // user's privacy settings. Otherwise, we consider the request to
     // be for an "anonymous" token.
 
     // TODO: What if these calls on userService have rejected promises?
+    var authorizationPromise = gpii.oauth2.getAuthorization(request.req, authGrantFinder);
+    var gpiiTokenHasAssociatedUserPromise = userService.gpiiTokenHasAssociatedUser(userToken);
+
+    var sources = [authorizationPromise, gpiiTokenHasAssociatedUserPromise];
+    var promisesSequence = fluid.promise.sequence(sources);
+
+    promisesSequence.then(function (responses) {
+        var authorization = responses[0];
+        var hasAssociatedUser = responses[1];
+
+        if (!authorization || authorization.gpiiToken !== userToken || !authorization.allowUntrustedSettings) {
```

Comment:
I would invert this logic and instead of checking the cases that cause the request to fail authorization, I would check the requirements for authorization and fail if those are not met (whitelist vs blacklist).

That is:

    if (authorization && authorization.gpiiToken === userToken && authorization.allowUntrustedSettings) {
        ...
    } else {
        // Reject...
    }